etna/transforms/math/power.py

```diff
@@ -61,6 +78,36 @@ def __init__(
             mode=mode,
         )
 
+    def _fit(self, df: pd.DataFrame) -> "SklearnTransform":
+        """
+        Fit transformer with data from df.
+
+        Here we are going to make a `patch <https://github.com/scikit-learn/scikit-learn/pull/23654>`_.
+        """
+        new_check_input = _replace_warnings_decorator(PowerTransformer._check_input)
```

Comment:
I think that we make it not correctly - `_replace_warnings_decorator` change sklearn behaviour completly.
Could you check the `np.warning` after `fit` execution?

Reply:
> How changing np.warnings = warnings could change the behavior completely?

Because np.warnings and warnings are different. We change global state without returning it back.
If it's okay why not just make it in  `__init__.py` - I don't see any reason to comlicate things with context managers and decorators if they do not return the global state to previos one.